### DISCUSSION TITLE
Build Error on windows Error: EPERM: operation not permitted, scandir 'C:\Users\admin\Application Data'

### DISCUSSION THREAD

**Discussion by vatoer** [vercel/next.js#62281] [Help]
### Summary

I need to build my project on Windows 11

I run 

```sh
pnpm build
```
and got error

```
 next build

   ▲ Next.js 14.1.0
   - Environments: .env

   Creating an optimized production build ...
glob error [Error: EPERM: operation not permitted, scandir 'C:\Users\admin\Application Data'] {
  errno: -4048,
  code: 'EPERM',
  syscall: 'scandir',
  path: 'C:\\Users\\admin\\Application Data'
}
Failed to compile.

Error: EPERM: operation not permitted, scandir 'C:\Users\admin\Application Data'


> Build failed because of webpack errors
 ELIFECYCLE  Command failed with exit code 1.
```

how to solve it? thanks

### Additional information

```js
## environment 

- windows 11
- Next.js 14.1.0
- pnpm 8.9.0

on Windows 11 it should be 'C:\Users\admin\AppData\ right ? 

so how to change  'C:\Users\admin\Application Data' to 'C:\Users\admin\AppData\ ?

Already tried with npm and also yarn, it gives same error.
```

**Comment by icyJoseph** [comment]
I think this is because of missing permissions? Like the build command is not allowed to write to that directory. 

I am not sure if in Windows Application Data requires some kind of admin level access. Try to move the project to Documents or something like that, that'd require less privileges.

**Reply by vatoer** [reply]
I'm using admin account, 

I find the culprit, 

its because my Prisma setting, I'm using custom location, I'll try to post in Prisma discussion 

just in case someone find same problem 

on my `schema.prisma` 

./prisma/my-db/schema.prisma

```prisma
generator client {
  provider = "prisma-client-js"
  output   = "  output   = "./generated/client""
}
```

**Reply by rhiskey** [reply]
> I'm using admin account,
> 
> I find the culprit,
> 
> its because my Prisma setting, I'm using custom location, I'll try to post in Prisma discussion
> 
> just in case someone find same problem
> 
> on my `schema.prisma`
> 
> ./prisma/my-db/schema.prisma
> 
> ```prisma
> generator client {
>   provider = "prisma-client-js"
>   output   = "  output   = "./generated/client""
> }
> ```

Just about half-year of waiting I ended up with cutting out this feature. Just don't use outputs with Prisma if you wish to build locally. I don't know why Prisma cant get access to those folders and why it is requested.

**Reply by itsnur-co** [reply]
> I'm using admin account,
> 
> I find the culprit,
> 
> its because my Prisma setting, I'm using custom location, I'll try to post in Prisma discussion
> 
> just in case someone find same problem
> 
> on my `schema.prisma`
> 
> ./prisma/my-db/schema.prisma
> 
> ```prisma
> generator client {
>   provider = "prisma-client-js"
>   output   = "  output   = "./generated/client""
> }
> ```

worked for me

**Reply by yf-hk** [reply]
> > I'm using admin account,
> > I find the culprit,
> > its because my Prisma setting, I'm using custom location, I'll try to post in Prisma discussion
> > just in case someone find same problem
> > on my `schema.prisma`
> > ./prisma/my-db/schema.prisma
> > ```prisma
> > generator client {
> >   provider = "prisma-client-js"
> >   output   = "  output   = "./generated/client""
> > }
> > ```
> 
> worked for me

Are you using Mac/Linux or Windows? It only happens on Windows

**Comment by karuppusamy-d** [comment]
Adding generated prisma output directory to `experimental.outputFileTracingIgnores` in next config.js solves the issue. 

next config.js
```js
experimental: {
    outputFileTracingIgnores: ["./generated/client/**/*"],
}
```

**Reply by doiali** [reply]
> Adding generated prisma output directory to `experimental.outputFileTracingIgnores` in next config.js solves the issue.
> 
> next config.js
> 
> ```js
> experimental: {
>     outputFileTracingIgnores: ["./generated/client/**/*"],
> }
> ```

I used this fix and it made some progress. Now the error has changed to this:
```
Error: EPERM: operation not permitted, scandir 'C:\Users\admin\Cookies'
```

I can't understand why!

**Reply by karuppusamy-d** [reply]
Update prisma output directory to '../node_modules/.prisma/client'

**Reply by Aibd** [reply]
You are really a genius

**Reply by pradip-vadher-34** [reply]
`generator client {
  provider = "prisma-client-js"
  output   = "../node_modules/.prisma/client"
}`

added like this but still having same error
`glob error [Error: EPERM: operation not permitted, scandir 'C:\Users\<user-name>\Cookies'] {
  errno: -4048,
  code: 'EPERM',
  syscall: 'scandir',
  path: 'C:\\Users\\<user-name>\\Cookies'
}`

**Reply by dimaslanjaka** [reply]
not work

**Comment by zchdoc** [comment]
Delete "import os from 'os'" and other imports related to system operations.

or delete the fold : C:\Users\admin\Application Data   

the fold size is zero

but all the unread zero size file or fold may ouput the error.....


i delete the code

// import os from 'os'
....
// const homeDir = os.homedir()

now It works well ，again

**Reply by cupid20103** [reply]
This is incorrect, I think

**Reply by auziqni** [reply]
no it doesnt work
i try this on fresh app, no import except prisma
i dont even have folder "Application Data"

**Reply by cupid20103** [reply]
On the schema.prisma, please remove the output option and write code like this

```
generator client {
  provider = "prisma-client-js"
}
```

If so, your code works well. I think

**Reply by dikshantsingh510** [reply]
Thanks, it works for me.
And update prisma client import from "@prisma/client" in **lib/prisma.ts** file. like 👇👇👇


 ```// lib/prisma.ts
import { PrismaClient } from "@prisma/client";
```

**Reply by gilmartatho** [reply]
Thanks for all.

**Comment by AzizOzenir** [comment]
When I moved project to c: disk, removed .next and node_modules folder and reinstall packages then it fixed

**Reply by abidali671** [reply]
Thanks, it's also worked for me.

**Reply by faridulhaque** [reply]
Thanks. Worked for me too.

**Reply by JaeHye0k** [reply]
Thanks, it's worked for me. 
Does anyone happen to know what was the exact problem?

**Reply by yf-hk** [reply]
Well, this is clearly a Next.js bug. Their poor code quality and the usage of weback worries me.

**Reply by inuka-mayakaduwa** [reply]
Worked for me, thankyou !

**Reply by Kimlong168** [reply]
it works for me

**Reply by mohammad-khoddami** [reply]
It works fine.

**Reply by AzizOzenir** [reply]
> Well, this is clearly a Next.js bug. Their poor code quality and the usage of weback worries me.

its probable potential cause.

**Reply by mohsn-mirzaei** [reply]
Thanks, it's also worked for me.

**Comment by devSubham35** [comment]
I faced the same issue and here’s how I ✅ solved it 👇
⚠️ Note: After doing this changes please remove .next folder 
and run prisma command ```yarn prisma generate``` then yarn build (in my case i use yarn)

Solution 1 -> Basically, the Prisma output generated file path should be changed to `"/node_modules/prisma/client"`.

```
/// In your schema.prisma file, update the generator block like this:

/// before
generator client {
  provider = "prisma-client-js"
  output   = "../lib/generated/prisma"
}

/// after
generator client {
  provider = "prisma-client-js"
  output   = "../node_modules/prisma/client"
}
```


Solution 2 -> In the prisma client instance file just change the import part
```
/// before
import { PrismaClient } from "./generated/prisma";

/// after
import { PrismaClient } from "@prisma/client";

const globalForPrisma = global as unknown as { prisma: PrismaClient };
export const prisma = globalForPrisma.prisma || new PrismaClient();
if (process.env.NODE_ENV !== "production") {
  globalForPrisma.prisma = prisma;
}
```
⚠️ Note: This issue only occurs on Windows, not on Linux.

**Reply by beautifulcoder** [reply]
Changing the Prisma client to `@prisma/client` works for me but this feels hacky as hell. At least I am able to get compiler feedback in local.

**Reply by paaauldev** [reply]
Thanks man! Working for me

**Comment by GeorgeeFlash** [comment]
I deleted `.next` & `node_modules`, ran `pnpm install`, yet I still got the error.
I deleted the folders again, changed directory for my project, ran `pnpm install`, still got the error.
I ran build as administrator, still got the error.
I removed output directory and got the following:
<img width="1362" height="332" alt="image" src="https://github.com/user-attachments/assets/acbe5dc5-2d8d-4399-8933-c53e57b61b65" />

😣
I deleted the generated prisma client output directory at `@/lib/generated` and the error went away.

**Reply by mohsn-mirzaei** [reply]
Move your source to the C: drive, and your problem will be solved
